Source/WTF/wtf/RunLoop.cpp

```diff
@@ -169,15 +170,16 @@ void RunLoop::dispatch(Function<void()>&& function)
         wakeUp();
 }
 
-void RunLoop::dispatchAfter(Seconds delay, Function<void()>&& function)
+Ref<RunLoop::DispatchTimer> RunLoop::dispatchAfter(Seconds delay, Function<void()>&& function)
 {
     RELEASE_ASSERT(function);
-    auto timer = new DispatchTimer(*this);
-    timer->setFunction([timer, function = WTFMove(function)] {
+    Ref<DispatchTimer> timer = adoptRef(*new DispatchTimer(*this));
+    timer->setFunction([timer = timer.copyRef(), function = WTFMove(function)] {
         function();
-        delete timer;
+        timer->stop();
```

Comment:
Let's also explicitly release the timer. Like,
```
Ref<DispatchTimer> protected { WTFMove(timer) };
function();
protected->stop();
```
Then, after the function call, protected will be released.

Reply:
This `timer` is `const`. Can I do `Ref<DispatchTimer> protectedTimer { timer.copyRef() };`?